### PR TITLE
Use Azure OpenAI service if configured

### PR DESCRIPTION
**Pull request by juberti**

### CODE REVIEW

File: packages/ai-jsx/src/lib/openai.tsx [hunk 1]
```diff
@@ -45,17 +45,45 @@ export type ValidChatModel =
...
+  protected override authHeaders() {
+    return {
+      'Api-Key': this.apiKey,
+    };
+  }
+  override buildRequest(options: FinalRequestOptions) {
+    const result = super.buildRequest(options);
+    if (options.body && 'model' in options.body) {
+      const model = (options.body.model as string).replace('.', '');
+      result.url = result.url.replace('/chat/', `/deployments/${model}/chat/`);
```
Comment (NickHeiner) [inline packages/ai-jsx/src/lib/openai.tsx:67]: For extra safety, you could use the [url](https://nodejs.org/api/url.html) module to parse out the `pathname` and only run the `replace` on that.

But I guess the `/` characters can only appear in the pathname, so this probably will not have an impact in practice.

**Review by NickHeiner — APPROVED** [review]
Neat that the diff to do this was so small!

This will be a minor version bump for the release.

**Review by mdwelsh — APPROVED** [review]

File: packages/ai-jsx/src/lib/openai.tsx (hunk 1 above)
Comment (juberti) [reply packages/ai-jsx/src/lib/openai.tsx:67]: yeah good call, I found a cleaner approach that isn't isolated to /chat/, so I went with that.

File: packages/ai-jsx/src/lib/openai.tsx [hunk 2]
```diff
@@ -45,17 +46,44 @@ export type ValidChatModel =
...
+  }
+}
+
 const openAiClientContext = AI.createContext<() => OpenAIClient>(
   _.once(() => {
     const baseURL = getEnvVar('OPENAI_API_BASE', false);
-    return new OpenAIClient({
-      apiKey: getEnvVar('OPENAI_API_KEY', false),
+    const useAzure = baseURL?.endsWith('azure.com');
+    const apiKey = getEnvVar(useAzure ? 'OPENAI_AZURE_API_KEY' : 'OPENAI_API_KEY', false);
```
Comment (petersalas) [inline packages/ai-jsx/src/lib/openai.tsx:77]: Should we fall back to `OPENAI_API_KEY` if `useAzure` and `OPENAI_AZURE_API_KEY` isn't set? Or maybe the OpenAI SDK will already do that?

**Review by petersalas — APPROVED** [review]

File: packages/ai-jsx/src/lib/openai.tsx (hunk 2 above)
Comment (juberti) [reply packages/ai-jsx/src/lib/openai.tsx:77]: I think we should error in that case (which the OAI SDK will do, I think)

**Pull request merged**